Packs/OpenAI/Integrations/OpenAIGPT/OpenAIGPT.py

```diff
@@ -0,0 +1,454 @@
+
+
+def extract_assistant_message(response: dict[str, Any]) -> str:
+    """
+        Extracts the assistant message from a response.
+        Returns:
+        The assistant message as a string.
+    """
+
+    choices: list = response.get('choices', [])
+    if not choices:
+        return_error("Could not retrieve message from response.")
+
+    message = choices[0].get('message', {})
+    if not message:
+        return_error("Could not retrieve message from response.")
+
+    response_content = message.get('content', '')
+    if not response_content:
+        return_error("Could not retrieve message from response.")
+
+    return response_content
+
+
+def get_email_parts(entry_id: str) -> tuple[List[dict[str, str]] | None, str | None, str | None, str | None]:
+    """
+        Extracts and parses the headers, text body, and HTML body from an .eml file identified by a given entry ID.
+
+        Args:
+        - entry_id (str): The unique identifier for the uploaded .eml file in the war room.
+
+        Returns:
+        - tuple[List[Dict[str, str]] | None, str | None, str | None]: A tuple containing three elements:
+            - headers (List[Dict[str, str]] | None): A list of dictionaries where each dictionary represents an email header.
+            - text_body (str | None): The plain text body of the email, if available.
+            - html_body (str | None): The HTML body of the email, if available.
+    """
+    if not entry_id:
+        DemistoException("Provide an entryId of an uploaded '.eml' file.")
+
+    get_file_path_res = demisto.getFilePath(entry_id)
+    file_path = get_file_path_res["path"]
+    file_name = get_file_path_res["name"]
+
+    if not file_name.endswith(EML_FILE_PREFIX):
+        DemistoException("Provided 'entry_id' does not point to a valid '.eml' file.")
+
+    email_parser = parse_emails.EmailParser(file_path=file_path)
+    email_parser.parse()
+
+    headers, text_body, html_body = (email_parser.parsed_email.get('Headers', None),
+                                     email_parser.parsed_email.get('Text', None),
+                                     email_parser.parsed_email.get('HTML', None))
+    return headers, text_body, html_body, file_name
+
+
+def check_email_part(email_part: str, client: OpenAiClient, args: dict[str, Any]) -> CommandResults:
+    """
+        Checks email parts (headers/body) for potential security issues using predefined prompts
+        ('CHECK_EMAIL_HEADERS_PROMPT', 'CHECK_EMAIL_BODY_PROMPT') that are sent to the GPT model.
+    """
+    entry_id: str = args.get(ArgAndParamNames.ENTRY_ID, '')
+    email_headers, email_text_body, email_html_body, file_name = get_email_parts(entry_id)
+    additional_instructions = f'Additional instructions: {ArgAndParamNames.ADDITIONAL_INSTRUCTIONS}\n'\
+        if args.get(ArgAndParamNames.ADDITIONAL_INSTRUCTIONS, "") else ''
+
+    if email_part == EmailParts.HEADERS:
+        demisto.debug(f'openai-gpt checking email headers: {email_headers=}')
```

Comment:
no check for header values and access to header values [0] is not in try/catch. 
I think need to add a check like you did in email body

Reply:
done